test/verify.hpp

```diff
@@ -77,8 +77,11 @@ struct not_finite_fn
     template <class T>
     bool operator()(T x) const
     {
-        using std::isfinite;
-        return not isfinite(x);
+        // WIN32: The Standard Library from MSVC does implement std::isfinite()
+        // for floating-point types only - no additional overloads for integer
+        // types, which should be treated as doubles according to the specification.
+        // https://en.cppreference.com/w/cpp/numeric/math/isfinite
+        return not std::isfinite(static_cast<double>(x));
```

Comment:
According to the specification, all integer types in additional overloads are treaded as doubles - reference https://en.cppreference.com/w/cpp/numeric/math/isnan. You do not provide a custom overload, e.g., for denormal, so here, the integer value will be implicitly converted to double, and then the floating-point version is called.

Reply:
The `double` still has enough spare bits in the exponent to represent all float denormalized numbers as normalized.

Reply:
Let's make it clear that this is a workaround

Reply:
The most significant disadvantage is existent `T` to `double` conversion - I don't think that it's a performance critical code but returning `true` for integers and call native functions for other types seems a bit better from performance point of view.

Also I've just realized that there is a one potential problem - if fp exceptions are turned on, we may get one during conversion. I don't think that MIOpen checks fpenv anywhere or turns on explicit fp exceptions, but the behavior is different for `std::isfinite(static_cast<double>(std::numeric_limits<uint64_t>::max()))` and `std::isfinite(std::numeric_limits<uint64_t>::max())`. It may alter fp state even if you work with integers.

Reply:
I guess we can discuss this problem and decide is it critical or not and probably put a few comments that this is fine.